internal/server/admin/key_service.go

```diff
@@ -0,0 +1,144 @@
+package admin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	pbModel "github.com/kong/koko/internal/gen/grpc/kong/admin/model/v1"
+	v1 "github.com/kong/koko/internal/gen/grpc/kong/admin/service/v1"
+	"github.com/kong/koko/internal/model"
+	"github.com/kong/koko/internal/resource"
+	"github.com/kong/koko/internal/server/util"
+	"github.com/kong/koko/internal/store"
+	"go.uber.org/zap"
+)
+
+type KeyService struct {
+	v1.UnimplementedKeyServiceServer
+	CommonOpts
+}
+
+func (s *KeyService) GetKey(
+	ctx context.Context,
+	req *v1.GetKeyRequest,
+) (*v1.GetKeyResponse, error) {
+	logger := s.logger(ctx)
+	db, err := s.getDB(ctx, req.Cluster)
+	if err != nil {
+		return nil, err
+	}
+	result := resource.NewKey()
+	err = getEntityByIDOrName(ctx, req.Id, result, store.GetByName(req.Id), db, logger)
+	if err != nil {
+		return nil, util.HandleErr(ctx, logger, err)
+	}
+
+	return &v1.GetKeyResponse{
+		Item: result.Key,
+	}, nil
+}
+
+func (s *KeyService) CreateKey(
+	ctx context.Context,
+	req *v1.CreateKeyRequest,
+) (*v1.CreateKeyResponse, error) {
+	db, err := s.getDB(ctx, req.Cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	res := resource.NewKey()
+	res.Key = req.Item
+	if err := db.Create(ctx, res); err != nil {
+		s.logger(ctx).Error("error creating", zap.Error(err))
```

Comment:
Can we have this be a full sentence? e.g.:
```suggestion
		s.logger(ctx).Error("unable to create new key entity", zap.Error(err))
```